### ISSUE TITLE
Context Menu Actions

### ISSUE THREAD

**Issue by MisterY** [belaviyo/keepass-macpass-helper#3]
Would be nice to have a context menu on a text field, with the same options as in the pop-up dialog:
- username
- password
- both

which would populate the fields. It would be nice if the form submission was configurable. Some forms require additional field, 2nd-step authentication data, or captcha so it is useful to have only the authentication fields populated but the form not being submitted.

**Comment by belaviyo** [comment]
> Would be nice to have a context menu on a text field, with the same options as in the pop-up dialog

This is not performance friendly. Note that for each site we will need to send a HTTP request to KeePassHTTP plugin and wait for the response. Right now this request is only being sent if user clicks on the toolbar button or uses the shortcut. For contextmenu, we will need to send requests beforehand so that all logins are available for listing in the context-menu.

>  but the form not being submitted

Open the panel and use shortcuts. For each action there are two shortcuts; one with auto-submit and one without.

**Comment by MisterY** [comment]
Hi, thanks for pointing this out. Unfortunately, neither Alt+B nor Ctrl+B type shortcuts work for me. The first shows the menu bar and the second opens the bookmark menu. Neither of them performs the auto-type in Firefox.
This leaves me only the option of clicking the option with a mouse, which is what I've been doing so far as I don't really want to have to move the hand to perform shortcuts constantly. The mouse-click option, however, performs the form submit and there is no option to get the fill-only function.

Regarding the context menu options - I'm wondering if it is possible to dynamically generate the sub-menus. In that case, the request would only be sent if the user explicitely opens the KeePassHelper menu.

**Comment by belaviyo** [comment]
> Unfortunately, neither Alt+B nor Ctrl+B type shortcuts work for me. 

Right! I just checked the extension on Windows OS and you are right. Ctrl + B is now supported on 0.2.0.

> I'm wondering if it is possible to dynamically generate the sub-menus.

To the best of my knowledge the answer is no. Take a look at https://developer.chrome.com/extensions/contextMenus for more info.

**Comment by gagarine** [comment]
> This is not performance friendly. 
This is the exact reason why I quit lastpass, adding stuff in webpage was killing Firefox performances.

**Comment by gamedevsam** [comment]
I recommend closing this issue, I find the current approach is a good compromise between performance and ease of use.
